test/sequential/test-worker-prof.js

```diff
@@ -19,20 +20,25 @@ if (process.argv[2] === 'child') {
   }
   const pingpong = `
   let counter = 0;
+  const fs = require('fs');
   const { Worker, parentPort  } = require('worker_threads');
   parentPort.on('message', (m) => {
-    if (counter++ === 1024)
+    if (counter++ === 10)
       process.exit(0);
      parentPort.postMessage(
-       m.toString().split('').reverse().toString().replace(/,/g, ''));
+       fs.readFileSync(m.toString()).toString());
```

Comment:
This turns the test from something that's CPU-bound to something that's I/O-bound. That mostly invalidates the test, IMO.

Another potential issue: slurping the binary like that is probably very slow on underpowered buildbots, like the Raspberry Pis.

Reply:
@bnoordhuis :

here is the history of this piece of code:

oirignally we were fully CPU bound:

https://github.com/nodejs/node/blob/fab06681236994c3ac77a1f259373fb9731ed6f7/test/parallel/test-worker-prof.js#L18-L19

but then this was failing, so modified to do a little I/O in between, to give profiler some CPU. that is where the current master is - that seem to fail in some windows systems.

https://github.com/nodejs/node/blob/d5674fd482249865abb24d3cbc39215d411cf444/test/sequential/test-worker-prof.js#L26-L28

and here we are now reading and sending the image content. Isn't it still a mix of CPU and I/O? May be with a little more towards I/O than towards CPU? In either way, as long as it is able to reliably allow the profiler to profile it, we are good? (this is a regression test for a bug by which there was very little tick for the worker)

> slurping the binary like that is probably very slow on underpowered buildbots, like the Raspberry Pis.

so we have reduced the iteration count from 1024 to 10, to reduce disc pressure.

In overall, the aim was to find a right mix of CPU + I/O that is characteristic of a workload, and will exercise the profiler in a reasonable manner. If there is a better load that can be tried here, we could try that as well!